### PR TITLE
Update Emergency Room Endpoint

### LINKED ISSUE #22: `UpdateEmergencyRoom`

### PR DESCRIPTION
**Pull request by use-to**
closes #22

### CODE REVIEW

File: routes/rooms.go [hunk 1]
```diff
@@ -127,12 +127,56 @@ func CreateEmergencyRoom(ctx *gin.Context) {
...
+// @Failure    400                             {object}    HTTPErrorResponse
 // @Router     /er/{id}                        [patch]
 func UpdateEmergencyRoom(ctx *gin.Context) {
-	_ = ctx.Param("id")
-	SendError(ctx, http.StatusNotImplemented, errors.New("this endpoint is not implemented yet"))
+	log, logCtx := logging.GetRequestLogger(ctx)
+	db := models.GetDB(logCtx)
+
+	erIdRaw := ctx.Param("id")
+	log.Debug().Str("update_id", erIdRaw)
```
Comment (FoseFx) [inline routes/rooms.go:139]: .Send is missing

File: routes/rooms.go [hunk 2]
```diff
@@ -127,12 +127,56 @@ func CreateEmergencyRoom(ctx *gin.Context) {
...
+	}
+
+	er := models.EmergencyRoom{
+		ID: erID,
+	}
+
+	// validate update body
+	body := PutERRequest{}
+	if err := ctx.ShouldBindJSON(&body); err != nil {
+		log.Warn().Err(err).Msg("validation failed")
```
Comment (FoseFx) [inline routes/rooms.go:153]: .Send missing

File: routes/rooms.go [hunk 3]
```diff
@@ -127,12 +127,56 @@ func CreateEmergencyRoom(ctx *gin.Context) {
...
+		SendError(ctx, http.StatusBadRequest, err)
+		return
+	}
+	log.Debug().Str("body", util.Formatted(body)).Send()
+
+	// convert department UUIDs into Departments
+	deps := make([]models.Department, len(body.Departments))
+	for i := range body.Departments {
```
Comment (FoseFx) [inline routes/rooms.go:161]: At this point we should write a function for this

**Review by FoseFx — CHANGES_REQUESTED** [review]

File: routes/rooms.go [hunk 4]
```diff
@@ -127,12 +127,56 @@ func CreateEmergencyRoom(ctx *gin.Context) {
 // @Produce    json
 // @Param      authorization                   header      string                true    "Bearer: <TOKEN>"
 // @Param      id                              path        string                true    "Emergency Room's ID"
-// @Success    200                             {object}    GetSingleERResponse
-// @Failure    501                             {object}    HTTPErrorResponse
+// @Param      emergency-room                  body        PutERRequest          true    "ER to update"
+// @Success    200
```
Comment (FoseFx) [inline routes/rooms.go:131]: Should we return the new state?

File: routes/rooms.go [hunk 5]
```diff
@@ -149,7 +193,7 @@ func DeleteEmergencyRoom(ctx *gin.Context) {
 	db := models.GetDB(logCtx)
 
 	erIdRaw := ctx.Param("id")
-	log.Debug().Str("requested_id", erIdRaw)
+	log.Debug().Str("delete_id", erIdRaw)
```
Comment (FoseFx) [inline routes/rooms.go:196]: .Send missing

File: routes/rooms.go (hunk 4 above)
Comment (use-to) [reply routes/rooms.go:131]: No, I think you should fetch it using the GET-endpoint.

File: routes/rooms.go (hunk 5 above)
Comment (use-to) [reply routes/rooms.go:196]: This is missing in every single endpoint? Why so?

Comment (FoseFx) [reply routes/rooms.go:196]: negligence

File: routes/rooms.go (hunk 3 above)
Comment (use-to) [reply routes/rooms.go]: fixed

File: routes/rooms.go (hunk 5 above)
Comment (use-to) [reply routes/rooms.go]: fixed

File: routes/rooms.go (hunk 1 above)
Comment (use-to) [reply routes/rooms.go]: fixed

File: models/department.go [hunk 6]
```diff
@@ -21,3 +21,13 @@ func DepartmentsToBases(deps []Department) []DepartmentBase {
 	}
 	return bases
 }
+
+// convert department UUIDs into Departments
```
Comment (FoseFx) [inline models/department.go:25]: I'm not sure why the linter does not fail, but [this is against conventions](https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences)

File: routes/rooms.go [hunk 7]
```diff
@@ -127,12 +119,50 @@ func CreateEmergencyRoom(ctx *gin.Context) {
...
+	log.Debug().Str("body", util.Formatted(body)).Send()
+
+	// create update model for gORM
+	updatedEr := models.EmergencyRoom{
+		EmergencyRoomBase: body.EmergencyRoomBase,
+		Departments:       models.UUIDsToDepartments(body.Departments),
+	}
+	log.Debug().Str("model", util.Formatted(er)).Send()
+
+	// acutally perform the update
```
Comment (FoseFx) [inline routes/rooms.go:158]: typo

**Review by FoseFx — APPROVED** [review]
other than that it's good to be merged

**Pull request merged**